_schemaV2.graphql

```diff
@@ -456,6 +456,12 @@ type AnalyticsShow {
   entityId: String!
 }
 
+# Statistics for users
+type AnalyticsUserStats {
+  numberOfPurchases: Int!
+  userId: String!
```

Comment:
Now that this `userId` is stitched here it looks more redundant. Also it has to be something like `internalID` to be consistent. So maybe we can remove it from vortex? @jpotts244 @starsirius @zephraph what do you think?

Reply:
Ah, like just remove it altogether since we are not reading the data from userId?

Reply:
**thought (no-strong-feeling):** I personally prefer including `userId` in the response so it's self-contained.

I know callers pass in `userId` as a param to get the user's stats, so they already have the `userId`. Sometimes I find it useful/convenient if the response also contains it so I can just pass the entire thing down without worrying the `userId` referenced somewhere else.

`AnalyticsUserStats` is stitched under `User` so it feels redundant as we are able to get the `userId` from `User`. But `AnalyticsUserStats` could also be used independently one day.

Re `internalID`, not en expert but I think `userId` might be preferable here because `AnalyticsUserStats` is not a user.

*format via [Conventional Comments](https://conventionalcomments.org/) + [saved replies](https://github.com/settings/replies)*

Reply:
Good point I agree `internalID` is not correct. With @starsirius point above that `userId` no longer bothers me that much 😄

Reply:
**nitpick**: `userId` -> `userID`

I agree that `internalID` wouldn't be good to add here in the same vein that @starsirius mentioned. `internalID` is meant to represent the id of that given type within the system that provides it. Often that's a database id. 

I'm fine with `userID` remaining.

Reply:
Don't want to open up this question now that we have an agreement 😄 but I tried this [here](https://github.com/artsy/metaphysics/pull/2586/files?file-filters%5B%5D=.ts#diff-7e804a266de6111194edd60c2104b605R49) to remove that `userId` when it is being stitched and it worked. Just throwing it here if you think it is good to make this change:
```graphql
type UserStats {
  totalPurchaseCount: Int!
}
extend type User {
  analytics: UserStats
}
```